app/models/shipit/task.rb

```diff
@@ -429,5 +434,11 @@ def duplicate_task
       copy_task.ended_at = nil
       copy_task
     end
+
+    def append_retry_if_necessary(event)
+      return event unless retries_configured?
+      return "#{event}_retry".to_sym if (status == "running" && retry_attempt > 0) || status == "timedout"
```

Comment:
Is this codepath only called when a Task is in the running state? That is, we may send Deploy/Task webhooks on creation, completion, etc. and so their status won't be "Running" in those cases.

Or is the intention to only differentiate a Deploy from a Retried deploy when it's a Running deploy event for a new retried deploy? (Oof, word salad.)

I think an approach like [my other comment](https://github.com/Shopify/shipit-engine/pull/1111/files#r509317438) would allow the receiver to make that differentiation if they so choose and remove the complexity here.

It's possible that some other webhook receivers may care to know if a failed or newly started deploy was also an automatic retry (e.g. I can envision some UI elements in ShipitNext to differntiate autodeploys like CD and retries from manual deploys).

Reply:
> the intention to only differentiate a Deploy from a Retried deploy when it's a Running deploy event for a new retried deploy

This^ is the intention. But you make a good point about this logic lying in the receiver as oppose to over here. So, I've taken it out and moved it to the spy PR 👍